File: src/repositories/summaryPrompt.repository.js
const { QueryTypes } = require('sequelize');
const sequelize = require('../config/db');

const findSummaryByGroupId = async (groupId) => {
    const sql = `
        SELECT sp.* FROM "SummaryPrompts" sp
        INNER JOIN "prompts" p ON sp."promptId" = p."id"
        WHERE p."groupId" = :groupId
        LIMIT 1
    `;
    
    const results = await sequelize.query(sql, {
        replacements: { groupId },
        type: QueryTypes.SELECT
    });

    return results[0] || null;
};

module.exports = {
    findSummaryByGroupId};

File: backend-java/src/main/java/com/voiceai/repository/SummaryPromptRepository.java
package com.voiceai.repository;

import com.voiceai.model.SummaryPrompt;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;
import java.util.UUID;

public interface SummaryPromptRepository extends JpaRepository<SummaryPrompt, UUID> {

    Optional<SummaryPrompt> findByPromptId(UUID promptId);

    @Query(value = """
            SELECT sp.*
            FROM "SummaryPrompts" sp
            JOIN prompts p ON sp."promptId" = p.id
            WHERE p."groupId" = :groupId
            LIMIT 1
            """, nativeQuery = true)
    Optional<SummaryPrompt> findByGroupId(UUID groupId);
}
